lambdas/pii-redact/src/pii-redact-handler.ts

```diff
@@ -0,0 +1,134 @@
+import { LambdaInterface } from "@aws-lambda-powertools/commons";
+import { Logger } from "@aws-lambda-powertools/logger";
+import * as zlib from "zlib";
+import { CloudWatchLogsDecodedData } from "aws-lambda";
+import { CloudWatchLogsEvent } from "aws-lambda/trigger/cloudwatch-logs";
+import {
+  CloudWatchLogsClient,
+  PutLogEventsCommand,
+  PutLogEventsCommandOutput,
+  CreateLogStreamCommand,
+} from "@aws-sdk/client-cloudwatch-logs";
+
+const logger = new Logger();
+const cloudwatch = new CloudWatchLogsClient();
+
+const ninoRegex = /\\"nino\\":\s*\\"([^"]*)\\"/g;
+const ipAddressRegex = /\\"ip_address\\":\s*\\"([^"]*)\\"/g;
+const userIdRegex = /\\"user_id\\":\s*\\"([^"]*)\\"/g;
+const firstNameRegex = /\\"firstName\\":\s*\\"([^"]*)\\"/g;
+const lastNameRegex = /\\"lastName\\":\s*\\"([^"]*)\\"/g;
+const birthDates =
+  /\\"birthDates\\"\s*:\s*\{\s*\\"L\\"\s*:\s*\[\s*\{\s*\\"M\\"\s*:\s*\{\s*\\"value\\"\s*:\s*\{\s*\\"S\\"\s*:\s*\\"(\d{4}-\d{2}-\d{2})\\"\s*}\s*}\s*}\s*]\s*}/g;
+const subjectRegex = /\\"subject\\":\s*\\"([^"]*)\\"/g;
+const tokenRegex = /\\"token\\":\s*\\"([^"]*)\\"/g;
+const dateOfBirthRegex = /\\"dateOfBirth\\":\s*\\"([^"]*)\\"/g;
+
+const buildingNameRegex =
+  /\\"buildingName\\"\s*:\s*{\s*\\"S\\"\s*:\s*\\"([^"]*)\\"/g;
+const addressLocalityRegex =
+  /\\"addressLocality\\"\s*:\s*{\s*\\"S\\"\s*:\s*\\"([^"]*)\\"/g;
+const buildingNumberRegex =
+  /\\"buildingNumber\\"\s*:\s*{\s*\\"S\\"\s*:\s*\\"([^"]*)\\"/g;
+const postalCodeRegex =
+  /\\"postalCode\\"\s*:\s*{\s*\\"S\\"\s*:\s*\\"([^"]*)\\"/g;
+const streetNameRegex =
+  /\\"streetName\\"\s*:\s*{\s*\\"S\\"\s*:\s*\\"([^"]*)\\"/g;
+
+const givenNameRegex =
+  /\\"type\\":{\\"S\\":\\"GivenName\\"},\s*\\"value\\":{\\"S\\":\\"([^"]*)\\"/g;
+const familyNameRegex =
```

Comment:
Maybe you could move this into it's own file and export it. Maybe you should move `redactPII(message: string)` into it own file then you can have test that just for the redacting bit of different string values and the outcomes